campaignion_newsletters/tests/ConditionalTest.php

```diff
@@ -20,44 +20,63 @@ public function setUp() {
+    $this->assertFalse($eq(['opt-in'], 'radios:no-change', $radios));
+    $this->assertFalse($eq(['opt-in'], 'radios:not-selected', $radios));
+    $this->assertFalse($eq(['opt-in'], 'checkbox:opt-in', $radios));
+    $this->assertFalse($eq(['opt-in'], 'checkbox:no-change', $radios));
+    $this->assertFalse($eq(['opt-out'], 'radios:opt-in', $radios));
+    $this->assertTrue($eq(['opt-out'], 'radios:opt-out', $radios));
+    $this->assertFalse($eq(['opt-out'], 'radios:no-change', $radios));
+    $this->assertFalse($eq(['opt-out'], 'radios:not-selected', $radios));
+    $this->assertFalse($eq(['opt-out'], 'checkbox:opt-in', $radios));
+    $this->assertFalse($eq(['opt-out'], 'checkbox:no-change', $radios));
+    $this->assertFalse($eq(['no-change'], 'radios:opt-in', $radios));
+    $this->assertFalse($eq(['no-change'], 'radios:opt-out', $radios));
+    $this->assertTrue($eq(['no-change'], 'radios:no-change', $radios));
+    $this->assertFalse($eq(['no-change'], 'radios:not-selected', $radios));
+    $this->assertFalse($eq(['no-change'], 'checkbox:opt-in', $radios));
+    $this->assertFalse($eq(['no-change'], 'checkbox:no-change', $radios));
+    $this->assertFalse($eq(['not-selected'], 'radios:opt-in', $radios));
+    $this->assertFalse($eq(['not-selected'], 'radios:opt-out', $radios));
+    $this->assertFalse($eq(['not-selected'], 'radios:no-change', $radios));
+    $this->assertTrue($eq(['not-selected'], 'radios:not-selected', $radios));
+    $this->assertFalse($eq(['not-selected'], 'checkbox:opt-in', $radios));
+    $this->assertFalse($eq(['not-selected'], 'checkbox:no-change', $radios));
+
+    $this->assertFalse($ne(['opt-in'], 'radios:opt-in', $radios));
+    $this->assertTrue($ne(['opt-in'], 'radios:opt-out', $radios));
+    $this->assertTrue($ne(['opt-in'], 'radios:no-change', $radios));
+    $this->assertTrue($ne(['opt-in'], 'radios:not-selected', $radios));
+    $this->assertTrue($ne(['opt-in'], 'checkbox:opt-in', $radios));
+    $this->assertTrue($ne(['opt-in'], 'checkbox:no-change', $radios));
+    $this->assertTrue($ne(['opt-out'], 'radios:opt-in', $radios));
+    $this->assertFalse($ne(['opt-out'], 'radios:opt-out', $radios));
+    $this->assertTrue($ne(['opt-out'], 'radios:no-change', $radios));
+    $this->assertTrue($ne(['opt-out'], 'radios:not-selected', $radios));
+    $this->assertTrue($ne(['opt-out'], 'checkbox:opt-in', $radios));
+    $this->assertTrue($ne(['opt-out'], 'checkbox:no-change', $radios));
+    $this->assertTrue($ne(['no-change'], 'radios:opt-in', $radios));
+    $this->assertTrue($ne(['no-change'], 'radios:opt-out', $radios));
+    $this->assertFalse($ne(['no-change'], 'radios:no-change', $radios));
+    $this->assertTrue($ne(['no-change'], 'radios:not-selected', $radios));
+    $this->assertTrue($ne(['no-change'], 'checkbox:opt-in', $radios));
+    $this->assertTrue($ne(['no-change'], 'checkbox:no-change', $radios));
+    $this->assertTrue($ne(['not-selected'], 'radios:opt-in', $radios));
```

Comment:
This is the test-case for testing the values directly provided by the form-API. So I guess it should still be a `[]` instead of `['not-selected']`.

Reply:
(Fixed that myself.)

Reply:
Your right! But this applies to the second block, testing what doesn't match, too, doesn't it?

Reply:
Yes, you’re right too :-)